external/fv3fit/tests/test_stacking.py

```diff
@@ -36,6 +37,26 @@ def test_StackedBatches_get_slice_stack():
         assert len(ds[SAMPLE_DIM_NAME]) == 20
 
 
+def test_StackedBatches_shuffles_before_thinning():
+    # This is a regression test which checks that when a sequence
+    # of batches with a "dataset" coordinate is used that the samples
+    # are stacked, shuffled, and then thinned (in that order).  Since
+    # we are setting the random seed, we can expect the same shuffling
+    # order every time.
+    data = np.arange(0, 36).reshape(2, 6, 3)
+    batch = xr.Dataset({"var": xr.DataArray(data, dims=["z", "x", DATASET_DIM_NAME],)})
+    stacked_batches = StackedBatches([batch, batch], np.random.RandomState(0))
+    stacked_batch = stacked_batches[0]
+
+    # If the order of shuffling and thinning were reversed, we would
+    # only select points from dataset zero, meaning stacked_batch.dataset
+    # would be all zeros.
+    expected_x = np.array([0, 3, 0, 3, 3, 1])
+    expected_dataset = np.array([1, 1, 2, 0, 2, 2])
+    np.testing.assert_equal(stacked_batch.x, expected_x)
+    np.testing.assert_equal(stacked_batch.dataset, expected_dataset)
```

Comment:
This does ensure shuffling happens before thinning, but is this really a test that it shuffles before it thins? This looks like a general regression test to me - if so you could update the function name to be more general, while keeping the comments that explain why this test is here.

No changes needed here, since regression is a pretty good test in this case. In the future, you could explicitly test that some operation happens before another by using mocks and asserting directly that e.g. the shuffle mock was called before the thinning mock. This would be more robust to other changes that could cause this test to fail.